site-src/concepts/api-overview.md

```diff
@@ -133,91 +150,94 @@ different relationships that Gateways and Routes can have:
   a single Route to control application exposure simultaneously across different
   IPs, load balancers, or networks.
 
-*In summary, Gateways select Routes and Routes control their exposure. When a
-Gateway selects a Route that allows itself to be exposed, then the Route will
-bind to the Gateway. When Routes are bound to a Gateway it means their
-collective routing rules are configured on the underlying load balancers or
-proxies that are managed by that Gateway. Thus, a Gateway is a logical
+*In summary, Routes attach to Gateways and Gateways choose which attachments to
+allow. When a Route tries to attach to a Gateway that does not prevent it, then
+the Route will bind to the Gateway. When Routes are bound to a Gateway it means
+their collective routing rules are configured on the underlying load balancers
+or proxies that are managed by that Gateway. Thus, a Gateway is a logical
 representation of a networking data plane that can be configured through
 Routes.*
 
-#### Route Selection
-
-A Gateway selects routes based on the Route metadata, specifically the kind,
-namespace, and labels of Route resources. Routes are actually bound to specific
-listeners within the Gateway so each listener has a `listener.routes` field
-which selects Routes by one or more of the following criterea:
-
-- **Label** - A Gateway can select Routes via labels that exist on the
-resource (similar to how Services select Pods via Pod labels).
-- **Kind** - A Gateway listener can only select a single type of Route
-resource. This could be an HTTPRoute, TCPRoute, or a custom Route type.
-- **Namespace** - A Gateway can also control from which Namespaces Routes can be
-selected via the `namespaces.from` field. It supports three possible values:
-    - `SameNamespace` is the default option. Only Routes in the same namespace
-      as this Gateway will be selected.
-    - `All` will select Routes from all Namespaces.
-    - `Selector` means that Routes from a subset of Namespaces selected by a
-      Namespace label selector will be selected by this Gateway. When `Selector`
-      is used then the `listeners.routes.namespaces.selector` field can be used
-      to specify label selectors. This field is not supported with `All` or
-      `SameNamespace`.
-
-The below Gateway will select all HTTPRoute resources with the `expose:
-prod-web-gw` across all Namespaces in the cluster.
+#### Route binding handshake
```

Comment:
Maybe this can be merged with this: https://github.com/kubernetes-sigs/gateway-api/pull/762/files#diff-bb2765a3020ed8962e5e749402a8ace4539c6218bdc621a98625ca19b2700e82R126-R150.